### ISSUE TITLE
Does not build against Qt 5.15 due to qt_wrap_cpp behaving differently

### ISSUE THREAD

**Issue by Vogtinator** [qtwebkit/qtwebkit#954]
With Qt 5.15, `Qt5CoreMacros.cmake` now defines `qt_wrap_cpp` itself, as an alias to `qt5_wrap_cpp`: https://doc.qt.io/qt-5/qtcore-cmake-qt5-wrap-cpp.html

This one is incompatible with the upstream CMake version though, so moc isn't run as expected and the build fails with ```/home/abuild/rpmbuild/BUILD/qtwebkit-5.212.0-alpha3/Tools/QtTestBrowser/webinspector.h:37: undefined reference to `vtable for WebInspector'```

Workaround: `set(QT_NO_CREATE_VERSIONLESS_FUNCTIONS ON)`

**Comment by annulen** [comment]
Thanks for heads up!

**Comment by annulen** [comment]
"This one is incompatible with the upstream CMake version though" - could you elaborate please?

**Comment by Vogtinator** [comment]
```qt5_wrap_cpp(<VAR> src_file1 [src_file2 ...]
             [TARGET target]
             [OPTIONS ...]
             [DEPENDS ...])```

vs.

```qt_wrap_cpp(resultingLibraryName DestName SourceLists ...)```

`QTestBrowser` uses `qt_wrap_cpp(WebKit QtTestBrowser_SOURCES webinspector.h)`, so it might have to be `qt5_wrap_cpp(QtTestBrowser_SOURCES webinspector.h TARGET WebKit)` instead?

**Comment by annulen** [comment]
I think this should do well:
```diff
diff --git a/Source/WebKit/PlatformQt.cmake b/Source/WebKit/PlatformQt.cmake
index 9aed9906e364..9c44d43fc0c6 100644
--- a/Source/WebKit/PlatformQt.cmake
+++ b/Source/WebKit/PlatformQt.cmake
@@ -289,10 +289,9 @@ if (ENABLE_GEOLOCATION)
 endif ()
 
 if (USE_QT_MULTIMEDIA)
-    qt_wrap_cpp(WebKit WebKit_SOURCES
-        qt/Api/qwebfullscreenvideohandler.h
-    )
     list(APPEND WebKit_SOURCES
+        qt/Api/qwebfullscreenvideohandler.h
+
         qt/WebCoreSupport/FullScreenVideoQt.cpp
     )
 endif ()
diff --git a/Tools/QtTestBrowser/CMakeLists.txt b/Tools/QtTestBrowser/CMakeLists.txt
index 19827fd02cac..e4c5d442d6ff 100644
--- a/Tools/QtTestBrowser/CMakeLists.txt
+++ b/Tools/QtTestBrowser/CMakeLists.txt
@@ -25,6 +25,7 @@ set(QtTestBrowser_SOURCES
     qttestbrowser.cpp
     urlloader.cpp
     utils.cpp
+    webinspector.h
     webpage.cpp
     webview.cpp
 )
@@ -44,10 +45,6 @@ set(QtTestBrowser_LIBRARIES
     ${STATIC_LIB_DEPENDENCIES}
 )
 
-qt_wrap_cpp(WebKit QtTestBrowser_SOURCES
-    webinspector.h
-)
-
 qt5_add_resources(QtTestBrowser_SOURCES
     QtTestBrowser.qrc
 )
```

**Comment by Vogtinator** [comment]
That's what I tried as well and it didn't work for some reason. I didn't do a clean rebuild though.

**Comment by annulen** [comment]
Weird, it should work with incremental build as well

**Comment by Vogtinator** [comment]
Tried again, now it works. Weird.

**Comment by annulen** [comment]
Fixed pushed as 41a05f881cf1af7bfe9197b2d833fdcd515e33dc
